01/solutions/06-write-tests-for-sanitization-func/src/lib_or.rs

```diff
@@ -0,0 +1,101 @@
+//! Run this file with `cargo test --test 06_write_tests`.
+
+/// This function implements a string sanitization logic that should uphold the following
+/// properties:
+/// - After sanitization, the result must not end with the character `x`
+/// - After sanitization, the result must not end with the character `o`
+/// - After sanitization, the result must not end with the string `.exe`
+///
+/// The function assumes that the input to the function only consists of lower and uppercase
+/// characters from the English alphabet and digits 0-9.
+///
+/// The implementation contains some bugs.
+///
+/// Your task is to write a set (at least 8) of unit tests, use them to find (at least 2) bugs in
+/// this function and then fix the function.
+/// 
+fn sanitize(input: &str) -> &str {
+    // Remove all x from the end of the string
+    let mut mark = true;
+    let mut input1 = input;
+
+    while mark && input1.len()>0 {
+        if input1.ends_with("x"){
+            mark = true;
+            input1 = input1.trim_end_matches("x");
+        }
+        else
+        if input1.ends_with("o"){
+            mark = true;
+            input1 = input1.trim_end_matches("o");
+        }
+        else
+        if input1.ends_with(".exe"){
+            mark = true;
+            input1 = &input1[0..input1.len() - 4];
+            // input1 = input1.trim_end_matches(".exe");
```

Comment:
This one is valid and easier to understand by code readers, so uncomment it and remove the above line.